tests/integration/repos_test.go

```diff
@@ -114,30 +114,55 @@ func TestRepositories_EditBranches(t *testing.T) {
 		t.Fatalf("Repositories.GetBranch() returned error: %v", err)
 	}
 
-	if *branch.Protection.Enabled {
+	if *branch.Protected {
 		t.Fatalf("Branch %v of repo %v is already protected", "master", *repo.Name)
 	}
 
-	branch.Protection.Enabled = github.Bool(true)
-	branch.Protection.RequiredStatusChecks = &github.RequiredStatusChecks{
-		EnforcementLevel: github.String("everyone"),
-		Contexts:         &[]string{"continous-integration"},
+	branch.Protected = github.Bool(true)
+
+	protectionRequest := &github.ProtectionRequest{
+		RequiredStatusChecks: &github.RequiredStatusChecks{
+			IncludeAdmins: github.Bool(true),
+			Strict:        github.Bool(true),
+			Contexts:      &[]string{"continuous-integration"},
+		},
+		Restrictions: &github.BranchRestrictionsRequest{
+			Users: &[]string{"u"},
+			Teams: &[]string{"t"},
+		},
 	}
-	branch, _, err = client.Repositories.EditBranch(*repo.Owner.Login, *repo.Name, "master", branch)
+
+	protection, _, err := client.Repositories.UpdateBranchProtection("o", "r", "b", protectionRequest)
 	if err != nil {
-		t.Fatalf("Repositories.EditBranch() returned error: %v", err)
+		t.Fatalf("Repositories.UpdateBranchProtection() returned error: %v", err)
 	}
 
-	if !*branch.Protection.Enabled {
-		t.Fatalf("Branch %v of repo %v should be protected, but is not!", "master", *repo.Name)
-	}
-	if *branch.Protection.RequiredStatusChecks.EnforcementLevel != "everyone" {
-		t.Fatalf("RequiredStatusChecks should be enabled for everyone, set for: %v", *branch.Protection.RequiredStatusChecks.EnforcementLevel)
+	if !*protection.RequiredStatusChecks.IncludeAdmins {
+		t.Fatalf("Branch %v of repo %v does not enforce the required status checks on the repository admins", "master", *repo.Name)
 	}
 
 	wantedContexts := []string{"continous-integration"}
-	if !reflect.DeepEqual(*branch.Protection.RequiredStatusChecks.Contexts, wantedContexts) {
-		t.Fatalf("RequiredStatusChecks.Contexts should be: %v but is: %v", wantedContexts, *branch.Protection.RequiredStatusChecks.Contexts)
+	if !reflect.DeepEqual(*protection.RequiredStatusChecks.Contexts, wantedContexts) {
+		t.Fatalf("RequiredStatusChecks.Contexts should be: %v but is: %v", wantedContexts, *protection.RequiredStatusChecks.Contexts)
+	}
+
+	want := &github.Protection{
+		RequiredStatusChecks: &github.RequiredStatusChecks{
+			IncludeAdmins: github.Bool(true),
+			Strict:        github.Bool(true),
+			Contexts:      &[]string{"continuous-integration"},
+		},
+		Restrictions: &github.BranchRestrictions{
+			Users: []*github.User{
+				{Login: github.String("u"), ID: github.Int(1)},
+			},
+			Teams: []*github.Team{
+				{Slug: github.String("t"), ID: github.Int(2)},
+			},
+		},
+	}
+	if !reflect.DeepEqual(protection, want) {
+		t.Errorf("Repositories.GetBranchProtection returned %+v, want %+v", protection, want)
 	}
```

Comment:
@shurcooL : The API is failing, 
I tried 
curl --user "username:password" -X PUT -d "" https://api.github.com/repos/username/test-6382800227808658932/branches/master/protection
{
  "message": "If you would like to help us test the Protected Branches API during its preview period, you must specify a custom media type in the 'Accept' header. Please see the docs for full details.",
  "documentation_url": "https://developer.github.com/v3/"
}

This is the same message when I run the code also. Is there something I am missing.

	repos_test.go:138: Repositories.UpdateBranchProtection() returned error: PUT https://api.github.com/repos/:userloginName/test-6382800227808658932/branches/master/protection: 422 Validation Failed [{Resource: Field: Code: Message:}]

The report_test.go file in GitHub folder passes. 
Is there something i am missing in the call.